.github/workflows/azure-remove-branch.yml

```diff
@@ -1,15 +1,71 @@
 name: Clean up Flexion Azure Resources
 
 on:
+  pull_request:
+    types: [ closed ]
   workflow_dispatch:
     inputs:
       hashId:
         description: "Hash id of target branch deployemnt"
-        default: ""
+        default: ''
         type: string
 
 jobs:
+  list:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+          environment: ${{ vars.AZURE_ENVIRONMENT }}
+      - name: List existing branch deployments
+        run: |
+          echo "List Azure resource groups of current branch deployment"
+          az group list --query "[?tags.branchName].{ Name:name Branch:tags.branchName HashId:tags.branchHashId }" -o table
+
+  check:
+    runs-on: ubuntu-latest
+    if: inputs.hashId != '' || github.event.action == 'closed'
```

Comment:
```
on:
  pull_request:
    types: [ closed ]
```
With that, isn't the second part of this condition always true?

Reply:
Had discussion and this is okay for now. Aware that there might be multiple pull requests for same branch that may get merged in and will remove a deployment branch that may still have work in progress. Will design for this later on if it becomes a problem.